src/sdk/namespace_fs.js

```diff
@@ -513,7 +549,37 @@ class NamespaceFS {
                 if (r.common_prefix) {
                     res.common_prefixes.push(r.key);
                 } else {
-                    res.objects.push(this._get_object_info(bucket, r.key, r.stat, 'null'));
+                    const obj_info_list = [];
+                    const obj_info = this._get_object_info(bucket, r.key, r.stat, 'null', true);
+                    if (!list_versions && obj_info.delete_marker) {
+                        continue;
+                    }
+                    if (obj_info.delete_marker) {
+                        obj_info.is_latest = false;
+                    }
+                    if (list_versions) {
+                        versions.forEach(v => {
+                            var search_key = v.key.replace(HIDDEN_VERSIONS_PATH, '');
+                            if (search_key.startsWith('/')) {
+                                search_key = search_key.replace('/', '');
+                            } else {
+                                search_key = search_key.replace('//', '/');
+                            }
+                            if (search_key.startsWith(r.key)) {
+                                obj_info_list.push(this._get_object_info(bucket, r.key, v.stat, 'null', false));
+                            }
+                        });
+                    }
+                    /* arrange the object versions in ascending create time */
```

Comment:
this means that S3 sorts the keys by name but the versions by creation time?

Reply:
Yes. I observed that the o/p was not sorted based on time. So sorting the entries based on c_time and pushing on the stack

Reply:
wait not sure I understand, in AWS it's not sorted by time but you did sort by time? 
according to what the versions should be sorted?

Reply:
`list_object_versions` list the objects based on the time, latest appears 1st in the o/p. We need to return the object_info sorted based on the time then we can match the behavior on AWS